cmd/notation/blob/inspect.go

```diff
@@ -0,0 +1,156 @@
+
+Example - Inspect BLOB signature and output as JSON:
+  notation blob inspect --output json <signature_path>
+`
+
+	command := &cobra.Command{
+		Use:   "blob inspect [signaturePath]",
+		Short: "Inspect signature associated with the signed BLOB",
+		Long:  longMessage,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing signature path to the artifact: use `notation blob inspect --help` to see what parameters are required")
+			}
+			opts.signaturePath = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBlobInspect(opts)
+		},
+	}
+
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	cmd.SetPflagOutput(command.Flags(), &opts.outputFormat, cmd.PflagOutputUsage)
+	return command
+}
+
+func runBlobInspect(opts *blobInspectOpts) error {
+	if opts.outputFormat != cmd.OutputJSON && opts.outputFormat != cmd.OutputPlaintext {
+		return fmt.Errorf("unrecognized output format %s", opts.outputFormat)
+	}
+
+	// initialize
+	mediaType, err := envelope.GetEnvelopeMediaType(filepath.Ext(opts.signaturePath))
+	if err != nil {
+		return err
+	}
+	contents, err := readFile(opts.signaturePath)
+	if err != nil {
+		return err
+	}
+	output := outputs.InspectOutput{MediaType: mediaType, Signatures: []outputs.SignatureOutput{}}
+	skippedSignatures := false
+	err, skippedSignatures, output.Signatures = outputs.Signature(mediaType, skippedSignatures, "nil", output, contents)
+	if err != nil {
+		return nil
+	}
+	if err := printOutput(opts.outputFormat, opts.signaturePath, output); err != nil {
+		return err
+	}
+	return nil
+}
+
+func printOutput(outputFormat string, ref string, output outputs.InspectOutput) error {
+	if outputFormat == cmd.OutputJSON {
+		return ioutil.PrintObjectAsJSON(output)
+	}
+
+	if len(output.Signatures) == 0 {
+		fmt.Printf("%s has no associated signature\n", ref)
+		return nil
+	}
+
+	root := tree.New(ref)
+	var signature outputs.SignatureOutput
+	root.AddPair("signature algorithm", signature.SignatureAlgorithm)
+	root.AddPair("signature envelope type", signature.MediaType)
+
+	signedAttributesNode := root.Add("signed attributes")
+	outputs.AddMapToTree(signedAttributesNode, signature.SignedAttributes)
+
+	unsignedAttributesNode := root.Add("unsigned attributes")
+	outputs.AddMapToTree(unsignedAttributesNode, signature.UnsignedAttributes)
+
+	certListNode := root.Add("certificates")
+	for _, cert := range signature.Certificates {
+		certNode := certListNode.AddPair("SHA256 fingerprint", cert.SHA256Fingerprint)
+		certNode.AddPair("issued to", cert.IssuedTo)
+		certNode.AddPair("issued by", cert.IssuedBy)
+		certNode.AddPair("expiry", cert.Expiry)
+	}
+
+	artifactNode := root.Add("signed artifact")
+	artifactNode.AddPair("media type", signature.SignedArtifact.MediaType)
+	artifactNode.AddPair("digest", signature.SignedArtifact.Digest.String())
+	artifactNode.AddPair("size", strconv.FormatInt(signature.SignedArtifact.Size, 10))
```

Comment:
This is duplicate of code in `cmd/notation/inspect.go`. Can we get rid of duplication?

Reply:
We can discuss this but i think we need it since we have two different outputs, the one for notation inspect is different from notation blob inspect and hence we need this

Reply:
Can you please share how output is different?

Reply:
Discussed offline, we can alter the output as per our need, will be working on these changes now